src/metricscollector/collector/apppoller_test.go

```diff
@@ -52,21 +52,31 @@ var _ = Describe("Apppoller", func() {
 			poller.Stop()
 		})
 
-		BeforeEach(func() {
-			cfc.GetTokensReturns(cf.Tokens{AccessToken: "test-access-token"})
+		It("polls metrics with the given interval", func() {
+			Eventually(noaa.ContainerMetricsCallCount).Should(Equal(1))
 
-			database.SaveMetricStub = func(metric *metrics.Metric) error {
-				Expect(metric.AppId).To(Equal("test-app-id"))
-				Expect(metric.Name).To(Equal(metrics.MetricNameMemory))
-				Expect(metric.Unit).To(Equal(metrics.UnitBytes))
-				Expect(metric.Instances).To(ConsistOf(metrics.InstanceMetric{Index: 0, Value: "1234"}))
-				return nil
-			}
+			fclock.Increment(TestPollInterval)
+			Eventually(noaa.ContainerMetricsCallCount).Should(Equal(2))
 
+			fclock.Increment(TestPollInterval)
+			Eventually(noaa.ContainerMetricsCallCount).Should(Equal(3))
 		})
 
 		Context("when retrieving container metrics all succeeds", func() {
 
+			BeforeEach(func() {
+				cfc.GetTokensReturns(cf.Tokens{AccessToken: "test-access-token"})
+
+				database.SaveMetricStub = func(metric *metrics.Metric) error {
```

Comment:
Normally you wouldn't do it this way since if SaveMetrics is never called these checks don't happen.
You would use the SaveMetricArgsForCall to retrieve the parameters and check in the test.